src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs

```diff
@@ -572,6 +572,51 @@ public static int Main ()
                 Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(5, 25));
         }
 
+        [Fact]
+        public void CS1035AtColonParsedAsComment_01()
+        {
+            var test = """
+                var x = @:;
+                """;
+
+            ParsingTests.ParseAndValidate(test,
+                // (1,9): error CS1056: Unexpected character '@'
+                // var x = @:;
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "@").WithArguments("@").WithLocation(1, 9),
+                // (1,12): error CS1733: Expected expression
+                // var x = @:;
+                Diagnostic(ErrorCode.ERR_ExpressionExpected, "").WithLocation(1, 12),
+                // (1,12): error CS1002: ; expected
+                // var x = @:;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(1, 12));
+        }
+
+        [Fact]
+        public void CS1035AtColonParsedAsComment_02()
+        {
+            var test = """
+                @:<div>test</div>
```

Comment:
Today in Razor you can "spill" this markup transition over multiple lines via something like this:

```razor
@:@{<strong>
    I'm evil
</strong>}
```

Will that still work with the single-line comment lexing?

Reply:
It will "work" in that the first line will entirely be processed as a comment, and the next line will be treated as an identifier. We'll have to have tests for this and handle it carefully on the razor side.